### ISSUE TITLE
Exception has occurred: ReferenceError: process is not defined

### ISSUE THREAD

**Issue by aalmada** [rarible/sdk#413]
I have a React project with a similar structure to the example from this repository. Unfortunately I'm getting this error when calling `createRaribleSdk` and I can't figure it why.

![image](https://user-images.githubusercontent.com/534533/210848897-3292071b-c852-45be-883b-b46fea59ceec.png)

![image](https://user-images.githubusercontent.com/534533/210848924-8c685988-fd16-4ff1-8fb3-5bddd68b4013.png)

I'm calling the method as shown at https://github.com/rarible/sdk/blob/9bcfc82d9cc7369a2d927eb91d6cdffc9712c0ef/packages/example/src/components/connector/sdk-connection-provider.tsx#L49

I already tried a simpler `createRaribleSdk(connectionState.connection.wallet, "testnet")` but I get the same error.

These are my dependencies: 

```
"@imtbl/imx-sdk": "^1.24.0",
"@rarible/api-client": "0.0.14-beta1",
"@rarible/connector": "^0.13.2",
"@rarible/connector-beacon": "^0.13.2",
"@rarible/connector-fcl": "^0.13.2",
"@rarible/connector-fortmatic": "^0.13.2",
"@rarible/connector-helper": "^0.13.2",
"@rarible/connector-iframe": "^0.13.2",
"@rarible/connector-immutablex-link": "^0.13.2",
"@rarible/connector-mew": "^0.13.2",
"@rarible/connector-phantom": "^0.13.2",
"@rarible/connector-portis": "^0.13.2",
"@rarible/connector-solflare": "^0.13.2",
"@rarible/connector-torus": "^0.13.2",
"@rarible/connector-walletconnect": "^0.13.2",
"@rarible/connector-walletlink": "^0.13.2",
"@rarible/sdk": "^0.13.2",
"@rarible/web3-ethereum": "^0.13.2",
"@tanstack/react-query": "^4.14.6",
"@tanstack/react-query-devtools": "^4.16.1",
"antd": "^5.0.1",
"assert": "^2.0.0",
"axios": "^1.1.3",
"buffer": "^6.0.3",
"clipboard-copy": "^4.0.1",
"crypto-browserify": "^3.12.0",
"https-browserify": "^1.0.0",
"mersenne-twister": "^1.1.0",
"observable-hooks": "^4.2.1",
"path-browserify": "^1.0.1",
"react": "^18.2.0",
"react-dom": "^18.2.0",
"react-oauth2-code-pkce": "^1.8.3",
"react-router-dom": "^6.4.3",
"siwe": "^1.1.6",
"stream-browserify": "^3.0.0",
"stream-http": "^3.2.0",
"tslib": "^2.4.1",
"url": "^0.11.0",
"web-vitals": "^3.0.4"
```

Do you have ideia of what's causing this?

**Comment by aalmada** [comment]
I'm looking into everything that's possible but still no solution...

I had already noticed previously that the connection observable was not updating. I experimented replacing [@rixio/react](https://www.npmjs.com/package/@rixio/react) by [observable-hooks](https://www.npmjs.com/package/observable-hooks) and the observable started working. I don't know if this is related...

I don't know if it's any incompatibility because I'm using newer versions of React and dependencies...

**Comment by aalmada** [comment]
I forgot to mention that I don't experience this errors when executing the `packages/example`.

Notice from the stack calls that the error always happens inside `Web3Ethereum.getChainId`. I keep trying multiple things but I keep getting the error.

**Comment by aalmada** [comment]
Trying to use a non-Ethereum wallet. When using Phantom, I get this error: 
![image](https://user-images.githubusercontent.com/534533/211069042-79d658a1-0634-422b-9739-40091743dfa4.png)

**Comment by ex1st0r** [comment]
add to your package.json resolution:
```  
"resolutions": {
    "//": "See https://github.com/facebook/create-react-app/issues/11773",
    "react-error-overlay": "6.0.9"
  },
```
and ensure that in your yarn.lock or package-lock.json the following version:
```react-error-overlay@6.0.9, react-error-overlay@^6.0.9:
  version "6.0.9"
  resolved "https://registry.yarnpkg.com/react-error-overlay/-/react-error-overlay-6.0.9.tgz#3c743010c9359608c375ecd6bc76f35d93995b0a"
  integrity sha512-nQTTcUu+ATDbrSD1BZHr5kgSD4oF8OFjxun8uAaL8RwPBacGBNPf/yAuVVdx17N8XNzRDMrZ9XcKZHCjPW+9ew==
```
[read more there](https://stackoverflow.com/questions/70368760/react-uncaught-referenceerror-process-is-not-defined)

**Comment by aalmada** [comment]
@aalmada Thanks! I'll try it

**Comment by aalmada** [comment]
Hi @ex1st0r!

I tried the proposed solutions from the link but still didn't work. It was a great tip anyway as I then suspected it was a polyfill issue. I'm using react 18 which uses webpack 5. I had already explicitly set the polyfill redirects in my `craco.config.ts` but it looks like it required some more that didn't show up as error messages. 

I now found about the [`node-polyfill-webpack-plugin`](https://www.npmjs.com/package/node-polyfill-webpack-plugin) which automatically sets all the polyfills. For future reference, my `craco.config.ts` look as follow:

``` typescript
// import webpack from "webpack";
import NodePolyfillPlugin from "node-polyfill-webpack-plugin";

// eslint-disable-next-line import/no-anonymous-default-export
export default {
	webpack: {
		configure: {
			entry: "./src/index.tsx",
		},
		plugins: {
			add: [
				new NodePolyfillPlugin({
					excludeAliases: ["console"],
				}),
			],
		},
	},
};
```

Sorry but I honestly tried many things but missed this one. Thanks for the help!
